source/server.go

```diff
@@ -106,10 +113,10 @@ func (s *Server) recvRecords(stream pb.SourceService_StreamServer) error {
 }
 
 func (s *Server) SendAck(position sdk.Position) error {
-	stream := s.stream.Load()
-	if stream == nil {
-		return fmt.Errorf("no stream is open")
+	for s.stream.Load() == nil {
+		<-s.StreamIsSetCh
 	}
+	stream := s.stream.Load()
 	err := (*stream).Send(&pb.Ack{AckPosition: position})
```

Comment:
If we load the stream and the connection JUST dropped, we could get an error here. In case that happens we should probably log the error and go back to loading the stream and waiting until it's established again.